Using @Inject Lang in the JSRValidator constructor doesn't work. Maven fail during execution of a "none related" tests (SwissKnifeTest).  If I remove the newly added construction from the JSRValidator class, the Maven process run with success.

maybe I do not use it correctly ?

###### JSRValidator

```
public static class JSRValidator implements Validator<Object> {

    private final Lang requestLanguage;

    @Inject
    public JSRValidator(Lang requestLanguage) {
        this.requestLanguage = requestLanguage;
    }

    @Override
    public void validate(Object value, String field, Context context) {
        if (value != null) {
            final ValidatorFactory validatorFactory = javax.validation.Validation.buildDefaultValidatorFactory();
            final javax.validation.Validator validator = validatorFactory.getValidator();
            final Set<javax.validation.ConstraintViolation<Object>> violations = validator.validate(value);
            final Locale localeToUse = (context.getAcceptLanguage() == null) ? Locale.getDefault() : Locale.forLanguageTag(context.getAcceptLanguage().split(",", 2)[0]);
            final Validation validation = context.getValidation();

            for (javax.validation.ConstraintViolation<Object> violation : violations) {
                final String violationMessage = validatorFactory.getMessageInterpolator().interpolate(
                        violation.getMessageTemplate(),
                        new NinjaContextMsgInterpolator(value, violation.getConstraintDescriptor()),
                        localeToUse
                );
                final ConstraintViolation constraintViolation = ConstraintViolation.create(violationMessage, violation.getInvalidValue());
                validation.addBeanViolation(new FieldViolation(violation.getPropertyPath().toString(), constraintViolation));
            }
        }
    }

    @Override
    public Class<Object> getValidatedType() {
        return Object.class;
    }
}
```

###### Maven

```
...
Running ninja.utils.ResultHandlerTest
Tests run: 8, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.013 sec
Running ninja.utils.SecretGeneratorTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.001 sec
Running ninja.utils.SwissKnifeTest

Running ninja.utils.SwissKnifeTest
22:02:39.014 [main] ERROR ninja.utils.SwissKnife - No converter found to convert class java.lang.StringBuffer. Returning null. You may want to extend the class.
22:02:39.015 [main] ERROR ninja.utils.SwissKnife - Cannot convert from java.lang.String to java.lang.Integer. Conversion failed with null
java.lang.reflect.InvocationTargetException: null
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_40]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_40]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_40]
        at java.lang.reflect.Method.invoke(Method.java:497) ~[na:1.8.0_40]
        at ninja.utils.SwissKnife.convert(SwissKnife.java:159) ~[classes/:na]
        at ninja.utils.SwissKnifeTest.testConvert(SwissKnifeTest.java:63) [test-classes/:na]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_40]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_40]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_40]
        at java.lang.reflect.Method.invoke(Method.java:497) ~[na:1.8.0_40]
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47) [junit-4.11.jar:na]
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) [junit-4.11.jar:na]
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44) [junit-4.11.jar:na]
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) [junit-4.11.jar:na]
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271) [junit-4.11.jar:na]
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70) [junit-4.11.jar:na]
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50) [junit-4.11.jar:na]
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238) [junit-4.11.jar:na]
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63) [junit-4.11.jar:na]
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236) [junit-4.11.jar:na]
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53) [junit-4.11.jar:na]
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229) [junit-4.11.jar:na]
        at org.junit.runners.ParentRunner.run(ParentRunner.java:309) [junit-4.11.jar:na]
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252) [surefire-junit4-2.12.4.jar:2.12.4]
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141) [surefire-junit4-2.12.4.jar:2.12.4]
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112) [surefire-junit4-2.12.4.jar:2.12.4]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_40]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_40]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_40]
        at java.lang.reflect.Method.invoke(Method.java:497) ~[na:1.8.0_40]
        at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189) [surefire-api-2.12.4.jar:2.12.4]
        at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165) [surefire-booter-2.12.4.jar:2.12.4]
        at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85) [surefire-booter-2.12.4.jar:2.12.4]
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115) [surefire-booter-2.12.4.jar:2.12.4]
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75) [surefire-booter-2.12.4.jar:2.12.4]
        Caused by: java.lang.NumberFormatException: For input string: "aString"
        at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65) ~[na:1.8.0_40]
        at java.lang.Integer.parseInt(Integer.java:580) ~[na:1.8.0_40]
        at java.lang.Integer.valueOf(Integer.java:766) ~[na:1.8.0_40]
        at ninja.utils.SwissKnife$Converter.toInteger(SwissKnife.java:180) ~[classes/:na]
        ... 35 common frames omitted
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec
Running ninja.utils.TimeUtilTest
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec

Results :

Tests in error:
(..)lidationWithNullObject(ninja.params.ControllerMethodInvokerTest): Guice configuration errors:
(..)lidationFailedWithThreeFields(ninja.params.ControllerMethodInvokerTest): Guice configuration errors:
(..)lidationFailedLength(ninja.params.ControllerMethodInvokerTest): Guice configuration errors:
(..)lidationFailedWithTwoAnnotations(ninja.params.ControllerMethodInvokerTest): Guice configuration errors:
(..)lidationPassed(ninja.params.ControllerMethodInvokerTest): Guice configuration errors:
(..)lidationFailedRange(ninja.params.ControllerMethodInvokerTest): Guice configuration errors:
(..)lidationFailedRegex(ninja.params.ControllerMethodInvokerTest): Guice configuration errors:

Tests run: 385, Failures: 0, Errors: 7, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] ninja .............................................. SUCCESS [  0.482 s]
[INFO] ninja-core ......................................... FAILURE [ 16.674 s]
[INFO] ninja-metrics ...................................... SKIPPED
[INFO] ninja-metrics-graphite ............................. SKIPPED
```

--- a/ninja-core/src/main/java/ninja/validation/Validators.java
+++ b/ninja-core/src/main/java/ninja/validation/Validators.java
@@ -16,7 +16,11 @@
 
 package ninja.validation;
 
+import com.google.common.base.Optional;
+import com.google.inject.Inject;
 import ninja.Context;
+import ninja.Result;
+import ninja.i18n.Lang;
 
 import javax.validation.MessageInterpolator;
 import javax.validation.ValidatorFactory;
@@ -63,11 +67,22 @@ public class Validators {
             this.descriptor = descriptor;
         }
 
+        /**
+         * Get the constraint descriptor.
+         *
+         * @return The constraint descriptor
+         * @see javax.validation.metadata.ConstraintDescriptor
+         */
         @Override
         public ConstraintDescriptor<?> getConstraintDescriptor() {
             return this.descriptor;
         }
 
+        /**
+         * Get the validated value.
+         *
+         * @return The value
+         */
         @Override
         public Object getValidatedValue() {
             return this.value;
@@ -76,16 +91,30 @@ public class Validators {
 
     public static class JSRValidator implements Validator<Object> {
 
+        private final Lang requestLanguage;
+
+        @Inject
+        public JSRValidator(Lang requestLanguage) {
+            this.requestLanguage = requestLanguage;
+        }
+
+        /**
+         * Validate the given value.
+         *
+         * @param value   The value, may be null
+         * @param field   The name of the field being validated, if applicable
+         * @param context The Ninja request context
+         */
         @Override
         public void validate(Object value, String field, Context context) {
             if (value != null) {
                 final ValidatorFactory validatorFactory = javax.validation.Validation.buildDefaultValidatorFactory();
                 final javax.validation.Validator validator = validatorFactory.getValidator();
                 final Set<javax.validation.ConstraintViolation<Object>> violations = validator.validate(value);
-                final Locale localeToUse = (context.getAcceptLanguage() == null) ? Locale.getDefault() : Locale.forLanguageTag(context.getAcceptLanguage().split(",", 2)[0]);
+                final Locale localeToUse = this.requestLanguage.getLocaleFromStringOrDefault(this.requestLanguage.getLanguage(context, Optional.<Result>absent()));
                 final Validation validation = context.getValidation();
 
-                for (javax.validation.ConstraintViolation<Object> violation : violations) {
+                for (final javax.validation.ConstraintViolation<Object> violation : violations) {
                     final String violationMessage = validatorFactory.getMessageInterpolator().interpolate(
                             violation.getMessageTemplate(),
                             new NinjaContextMsgInterpolator(value, violation.getConstraintDescriptor()),

--- a/ninja-core/src/test/java/ninja/params/ControllerMethodInvokerTest.java
+++ b/ninja-core/src/test/java/ninja/params/ControllerMethodInvokerTest.java
@@ -16,12 +16,30 @@
 
 package ninja.params;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Inject;
+import ninja.Context;
+import ninja.Result;
+import ninja.RoutingException;
+import ninja.i18n.Lang;
+import ninja.i18n.LangImpl;
+import ninja.session.FlashScope;
+import ninja.session.Session;
+import ninja.utils.NinjaMode;
+import ninja.utils.NinjaProperties;
+import ninja.utils.NinjaPropertiesImpl;
+import ninja.validation.*;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
 
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
 import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
@@ -30,63 +48,51 @@ import java.lang.reflect.Method;
 import java.util.ArrayList;
 import java.util.Arrays;
 
-import javax.validation.constraints.Max;
-import javax.validation.constraints.Min;
-import javax.validation.constraints.Pattern;
-import javax.validation.constraints.Size;
-
-import ninja.Context;
-import ninja.Result;
-import ninja.RoutingException;
-import ninja.session.FlashScope;
-import ninja.session.Session;
-import ninja.validation.JSR303Validation;
-import ninja.validation.NumberValue;
-import ninja.validation.Required;
-import ninja.validation.Validation;
-import ninja.validation.ValidationImpl;
-
-import org.junit.Before;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.mockito.Mock;
-import org.mockito.runners.MockitoJUnitRunner;
-
-import com.google.inject.AbstractModule;
-import com.google.inject.Guice;
-import com.google.inject.Inject;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
 
 
 @RunWith(MockitoJUnitRunner.class)
 public class ControllerMethodInvokerTest {
 
     @Mock
-	private MockController mockController;
+    private MockController mockController;
+
     @Mock
     private Context context;
+
     @Mock
     private Session session;
+
     @Mock
     private FlashScope flash;
+
+    private NinjaProperties ninjaProperties;
+
+    private Lang lang;
 
     private Validation validation;
 
     @Before
     public void setUp() throws Exception {
-        validation = new ValidationImpl();
+        this.ninjaProperties = new NinjaPropertiesImpl(NinjaMode.test);
+        this.lang = new LangImpl(ninjaProperties);
+        this.validation = new ValidationImpl();
+
         when(context.getSessionCookie()).thenReturn(session);
         when(context.getFlashCookie()).thenReturn(flash);
         when(context.getValidation()).thenReturn(validation);
     }
 
     @Test
-	public void noParameterMethodShouldBeInvoked() throws Exception {
-		create("noParameter").invoke(mockController, context);
+    public void noParameterMethodShouldBeInvoked() throws Exception {
+        create("noParameter").invoke(mockController, context);
         verify(mockController).noParameter();
-	}
+    }
 
-	@Test
-	public void contextShouldBePassed() throws Exception {
+    @Test
+    public void contextShouldBePassed() throws Exception {
         create("context").invoke(mockController, context);
         verify(mockController).context(context);
     }
@@ -105,11 +111,11 @@ public class ControllerMethodInvokerTest {
 
 
     @Test
-	public void paramAnnotatedArgumentShouldBePassed() throws Exception {
+    public void paramAnnotatedArgumentShouldBePassed() throws Exception {
         when(context.getParameter("param1")).thenReturn("value");
         create("param").invoke(mockController, context);
         verify(mockController).param("value");
-	}
+    }
 
     @Test
     public void pathParamAnnotatedArgumentShouldBePassed() throws Exception {
@@ -158,7 +164,7 @@ public class ControllerMethodInvokerTest {
     public void headersAnnotatedArgumentShouldBePassed() throws Exception {
         when(context.getHeaders("param1")).thenReturn(Arrays.asList("a", "b", "c"));
         create("headers").invoke(mockController, context);
-        verify(mockController).headers(new String [] {"a", "b", "c"});
+        verify(mockController).headers(new String[]{"a", "b", "c"});
     }
 
     @Test
@@ -216,7 +222,7 @@ public class ControllerMethodInvokerTest {
     public void shortParamShouldBeParsedToShort() throws Exception {
         when(context.getParameter("param1")).thenReturn("20");
         create("shortParam").invoke(mockController, context);
-        verify(mockController).shortParam((short)20);
+        verify(mockController).shortParam((short) 20);
     }
 
     @Test
@@ -244,7 +250,7 @@ public class ControllerMethodInvokerTest {
     @Test
     public void primShortParamShouldHandleNull() throws Exception {
         create("primShortParam").invoke(mockController, context);
-        verify(mockController).primShortParam((short)0);
+        verify(mockController).primShortParam((short) 0);
         assertFalse(validation.hasViolations());
     }
 
@@ -252,7 +258,7 @@ public class ControllerMethodInvokerTest {
     public void primShortValidationShouldWork() throws Exception {
         when(context.getParameter("param1")).thenReturn("blah");
         create("primShortParam").invoke(mockController, context);
-        verify(mockController).primShortParam((short)0);
+        verify(mockController).primShortParam((short) 0);
         assertTrue(validation.hasFieldViolation("param1"));
     }
 
@@ -288,7 +294,7 @@ public class ControllerMethodInvokerTest {
     public void byteParamShouldBeParsedToByte() throws Exception {
         when(context.getParameter("param1")).thenReturn("20");
         create("byteParam").invoke(mockController, context);
-        verify(mockController).byteParam((byte)20);
+        verify(mockController).byteParam((byte) 20);
     }
 
     @Test
@@ -542,7 +548,7 @@ public class ControllerMethodInvokerTest {
         ParamParsers.registerEnum(Rainbow.class);
         when(context.getParameter("param1")).thenReturn("Red");
         create("enumCsvParam").invoke(mockController, context);
-        verify(mockController).enumCsvParam(new Rainbow[] { Rainbow.Red });
+        verify(mockController).enumCsvParam(new Rainbow[]{Rainbow.Red});
         ParamParsers.unregisterEnum(Rainbow.class);
     }
 
@@ -551,7 +557,7 @@ public class ControllerMethodInvokerTest {
         ParamParsers.registerEnum(Rainbow.class);
         when(context.getParameter("param1")).thenReturn("Red,Orange,Yellow");
         create("enumCsvParam").invoke(mockController, context);
-        verify(mockController).enumCsvParam(new Rainbow[] { Rainbow.Red, Rainbow.Orange, Rainbow.Yellow});
+        verify(mockController).enumCsvParam(new Rainbow[]{Rainbow.Red, Rainbow.Orange, Rainbow.Yellow});
         ParamParsers.unregisterEnum(Rainbow.class);
     }
 
@@ -580,7 +586,7 @@ public class ControllerMethodInvokerTest {
         ParamParsers.registerEnum(Rainbow.class);
         when(context.getParameterValues("param1")).thenReturn(Arrays.asList("Blue"));
         create("enumArrayParam").invoke(mockController, context);
-        verify(mockController).enumArrayParam(new Rainbow[] { Rainbow.Blue });
+        verify(mockController).enumArrayParam(new Rainbow[]{Rainbow.Blue});
         ParamParsers.unregisterEnum(Rainbow.class);
     }
 
@@ -589,7 +595,7 @@ public class ControllerMethodInvokerTest {
         ParamParsers.registerEnum(Rainbow.class);
         when(context.getParameterValues("param1")).thenReturn(Arrays.asList("Blue", "Indigo", "Violet"));
         create("enumArrayParam").invoke(mockController, context);
-        verify(mockController).enumArrayParam(new Rainbow[] { Rainbow.Blue, Rainbow.Indigo, Rainbow.Violet});
+        verify(mockController).enumArrayParam(new Rainbow[]{Rainbow.Blue, Rainbow.Indigo, Rainbow.Violet});
         ParamParsers.unregisterEnum(Rainbow.class);
     }
 
@@ -701,8 +707,7 @@ public class ControllerMethodInvokerTest {
     public void validationPassed() {
         validateJSR303(buildDto("regex", "length", 5));
         assertFalse(context.getValidation().hasViolations());
-        assertFalse("Expected not to have regex violation.",
-                context.getValidation().hasBeanViolation("regex"));
+        assertFalse("Expected not to have regex violation.", context.getValidation().hasBeanViolation("regex"));
     }
 
     @Test
@@ -736,6 +741,24 @@ public class ControllerMethodInvokerTest {
                 context.getValidation().hasBeanViolation("range"));
         assertTrue(context.getValidation().getBeanViolations().get(0).field
                 .contentEquals("range"));
+    }
+
+    @Test
+    public void validationFailedTranslationFr() {
+        when(this.context.getAcceptLanguage()).thenReturn("fr");
+        validateJSR303(buildDto("regex", "length - too long", 5));
+        assertTrue(context.getValidation().hasViolations());
+        assertEquals(context.getValidation().getBeanViolations().size(), 1);
+        assertEquals(context.getValidation().getBeanViolations().get(0).constraintViolation.getMessageKey(), "la taille doit être entre 5 et 10");
+    }
+
+    @Test
+    public void validationFailedTranslationEn() {
+        when(this.context.getAcceptLanguage()).thenReturn("en");
+        validateJSR303(buildDto("regex", "length - too long", 5));
+        assertTrue(context.getValidation().hasViolations());
+        assertEquals(context.getValidation().getBeanViolations().size(), 1);
+        assertEquals(context.getValidation().getBeanViolations().get(0).constraintViolation.getMessageKey(), "size must be between 5 and 10");
     }
 
     @Test
@@ -786,12 +809,12 @@ public class ControllerMethodInvokerTest {
 
     private void validateJSR303(Dto dto) {
         when(context.parseBody(Dto.class)).thenReturn(dto);
-        create("JSR303Validation").invoke(mockController, context);
+        create("JSR303Validation", this.ninjaProperties, this.lang).invoke(mockController, context);
     }
 
     private void validateJSR303WithRequired(Dto dto) {
         when(context.parseBody(Dto.class)).thenReturn(dto);
-        create("JSR303ValidationWithRequired").invoke(mockController, context);
+        create("JSR303ValidationWithRequired", this.ninjaProperties, this.lang).invoke(mockController, context);
     }
 
     private Dto buildDto(String regex, String length, int range) {
@@ -820,54 +843,113 @@ public class ControllerMethodInvokerTest {
         }));
     }
 
+    public enum Rainbow {
+        Red, Orange, Yellow, Green, Blue, Indigo, Violet
+    }
+
+    // Custom argument extractors for testing different instantiation paths
+
     public interface MockController {
         public Result noParameter();
+
         public Result context(Context context);
+
         public Result session(Session session);
+
         public Result flash(FlashScope flash);
+
         public Result param(@Param("param1") String param1);
+
         public Result pathParam(@PathParam("param1") String param1);
+
         public Result sessionParam(@SessionParam("param1") String param1);
+
         public Result attribute(@Attribute("param1") Dep param1);
+
         public Result header(@Header("param1") String param1);
-        public Result headers(@Headers("param1") String [] param1);
+
+        public Result headers(@Headers("param1") String[] param1);
+
         public Result integerParam(@Param("param1") Integer param1);
+
         public Result intParam(@Param("param1") int param1);
+
         public Result shortParam(@Param("param1") Short param1);
+
         public Result primShortParam(@Param("param1") short param1);
+
         public Result characterParam(@Param("param1") Character param1);
+
         public Result charParam(@Param("param1") char param1);
+
         public Result byteParam(@Param("param1") Byte param1);
+
         public Result primByteParam(@Param("param1") byte param1);
+
         public Result booleanParam(@Param("param1") Boolean param1);
+
         public Result primBooleanParam(@Param("param1") boolean param1);
+
         public Result longParam(@Param("param1") Long param1);
+
         public Result primLongParam(@Param("param1") long param1);
+
         public Result floatParam(@Param("param1") Float param1);
+
         public Result primFloatParam(@Param("param1") float param1);
+
         public Result doubleParam(@Param("param1") Double param1);
+
         public Result primDoubleParam(@Param("param1") double param1);
+
         public Result enumParam(@Param("param1") Rainbow param1);
-        public Result enumCsvParam(@Param("param1") Rainbow [] param1);
-        public Result enumArrayParam(@Params("param1") Rainbow [] param1);
+
+        public Result enumCsvParam(@Param("param1") Rainbow[] param1);
+
+        public Result enumArrayParam(@Params("param1") Rainbow[] param1);
+
         public Result noArgArgumentExtractor(@NoArg String param1);
+
         public Result classArgArgumentExtractor(@ClassArg String param1);
+
         public Result guiceArgumentExtractor(@GuiceAnnotation(foo = "bar") String param1);
+
         public Result multiple(@Param("param1") String param1, @PathParam("param2") int param2,
-                Context context, Session session);
+                               Context context, Session session);
+
         public Result required(@Param("param1") @Required String param1);
+
         public Result requiredInt(@Param("param1") @Required @NumberValue(min = 10) int param1);
+
         public Result badValidator(@Param("param1") @NumberValue(min = 10) String param1);
+
         public Result body(Object body);
+
         public Result tooManyBodies(Object body1, Object body2);
 
         public Result JSR303Validation(@JSR303Validation Dto dto, Validation validation);
 
-        public Result JSR303ValidationWithRequired(@Required @JSR303Validation Dto dto,
-                Validation validation);
+        public Result JSR303ValidationWithRequired(@Required @JSR303Validation Dto dto, Validation validation);
     }
 
-    // Custom argument extractors for testing different instantiation paths
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target(ElementType.PARAMETER)
+    @WithArgumentExtractor(NoArgArgumentExtractor.class)
+    public @interface NoArg {
+    }
+
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target(ElementType.PARAMETER)
+    @WithArgumentExtractor(ClassArgArgumentExtractor.class)
+    public @interface ClassArg {
+    }
+
+    @Retention(RetentionPolicy.RUNTIME)
+    @Target(ElementType.PARAMETER)
+    @WithArgumentExtractor(GuiceArgumentExtractor.class)
+    public @interface GuiceAnnotation {
+        String foo();
+    }
 
     public static class NoArgArgumentExtractor implements ArgumentExtractor<String> {
         @Override
@@ -885,11 +967,6 @@ public class ControllerMethodInvokerTest {
             return null;
         }
     }
-
-    @Retention(RetentionPolicy.RUNTIME)
-    @Target(ElementType.PARAMETER)
-    @WithArgumentExtractor(NoArgArgumentExtractor.class)
-    public @interface NoArg {}
 
     public static class ClassArgArgumentExtractor implements ArgumentExtractor<String> {
         private final Class<?> clazz;
@@ -911,23 +988,6 @@ public class ControllerMethodInvokerTest {
         @Override
         public String getFieldName() {
             return null;
-        }
-    }
-
-    @Retention(RetentionPolicy.RUNTIME)
-    @Target(ElementType.PARAMETER)
-    @WithArgumentExtractor(ClassArgArgumentExtractor.class)
-    public @interface ClassArg {}
-
-    public class Dep {
-        private final String value;
-
-        public Dep(String value) {
-            this.value = value;
-        }
-
-        public String value() {
-            return value;
         }
     }
 
@@ -964,26 +1024,30 @@ public class ControllerMethodInvokerTest {
         }
     }
 
-    @Retention(RetentionPolicy.RUNTIME)
-    @Target(ElementType.PARAMETER)
-    @WithArgumentExtractor(GuiceArgumentExtractor.class)
-    public @interface GuiceAnnotation {
-        String foo();
+    public class Dep {
+
+        private final String value;
+
+        public Dep(String value) {
+            this.value = value;
+        }
+
+        public String value() {
+            return value;
+        }
     }
 
     public class Dto {
+
         @Size(min = 1, max = 10)
         @Pattern(regexp = "[a-z]*")
         public String regex;
+
         @Size(min = 5, max = 10)
         public String length;
+
         @Min(value = 3)
         @Max(value = 10)
         public int range;
-    }
-
-
-    public enum Rainbow {
-        Red, Orange, Yellow, Green, Blue, Indigo, Violet
     }
 }

